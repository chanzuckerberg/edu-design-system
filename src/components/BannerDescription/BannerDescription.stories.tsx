import { Story, Meta } from '@storybook/react';
import React from 'react';

import { BannerDescription, Props } from './BannerDescription';

export default {
  title: 'Molecules/Messaging/BannerDescription',
  component: BannerDescription,
} as Meta;

const Template: Story<Props> = (args) => <BannerDescription {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Your courses have been successfully updated.',
};
