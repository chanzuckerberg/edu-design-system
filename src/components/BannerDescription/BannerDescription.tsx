import React, { ReactNode } from 'react';
import Text from '../Text';

export interface Props {
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  /**
   * Children content
   */
  children: ReactNode;
}

/**
 * Used for the title text in the Banner.
 */
export const BannerDescription = ({ children, ...other }: Props) => {
  return <Text {...other}>{children}</Text>;
};
