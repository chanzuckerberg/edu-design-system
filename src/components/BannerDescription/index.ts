export { BannerDescription as default } from './BannerDescription';
