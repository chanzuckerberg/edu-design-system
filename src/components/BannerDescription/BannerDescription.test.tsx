import { generateSnapshots } from '@chanzuckerberg/story-utils';
import * as stories from './BannerDescription.stories';

describe('<BannerDescription />', () => {
  generateSnapshots(stories);
});
