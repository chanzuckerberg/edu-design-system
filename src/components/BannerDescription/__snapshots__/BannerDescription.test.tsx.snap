// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerDescription /> Default story renders snapshot 1`] = `
<p
  class="text text--body"
>
  Your courses have been successfully updated.
</p>
`;
